strawberryfields/tdm/tdmprogram.py

```diff
@@ -572,6 +576,38 @@ def unroll(self, shots):
             self.circuit = self.unrolled_circuit * shots
             return self
 
+        if self._is_space_unrolled:
+            raise ValueError("Program is space-unrolled and must be rolled before unrolling")
+
+        return self._unroll_program(shots)
+
+    def space_unroll(self, shots=1):
+        """Construct the space-unrolled program
+
+        Constructs the space-unrolled single-shot program, storing it in `self.space_unrolled_circuit`
```

Comment:
Based on this comment, I'd expect to find a line that is `self.space_unrolled_circuit = <somehow unrolling the circuit>`. Since this is not the case, maybe it's worth rephrasing this part a bit.